### PR TITLE
adding 3rd bunch of nf-core deps from update/nextflow-tools

### PR DESCRIPTION
**Pull request by marcodelapierre**
Fifth chunk of updates from #34466.

@adamjstewart

Note you had started to review these recipes in the PR above.

Based off #34562 : Review AFTER #34562

Review BEFORE #34564

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed non-darwin sha256 for all of the packages (i.e., did not confirm the darwin sha256 for `py-lagom`.

**Comment by tldahlgren** [comment]
Adding @adamjstewart since these are python packages.

**Comment by adamjstewart** [comment]
You'll need to resolve the merge conflicts from the other PR

**Comment by marcodelapierre** [comment]
OK @adamjstewart , good to go

File: var/spack/repos/builtin/packages/py-dnspython/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """DNS toolkit"""
+
+    homepage = "https://www.dnspython.org"
+    pypi = "dnspython/dnspython-2.2.1.tar.gz"
+
+    version("2.2.1", sha256="0f7569a4a6ff151958b64304071d370daa3243d15941a7beedf0c9fe5105603e")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools@44:", type="build")
+    depends_on("py-poetry", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnspython/package.py]:
```suggestion
    depends_on("py-poetry-core", type="build")
```

File: var/spack/repos/builtin/packages/py-dnspython/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class PyDnspython(PythonPackage):
+    """DNS toolkit"""
+
+    homepage = "https://www.dnspython.org"
+    pypi = "dnspython/dnspython-2.2.1.tar.gz"
+
+    version("2.2.1", sha256="0f7569a4a6ff151958b64304071d370daa3243d15941a7beedf0c9fe5105603e")
+
+    depends_on("python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnspython/package.py]:
```suggestion
    depends_on("python@3.6:", type=("build", "run"))
```
Or just remove it, Spack no longer supports older Python anyway

File: var/spack/repos/builtin/packages/py-dnspython/package.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("py-setuptools@44:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnspython/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+class PyFastapiUtils(PythonPackage):
+    """Reusable utilities for FastAPI"""
+
+    homepage = "https://fastapi-utils.davidmontague.xyz"
+    pypi = "fastapi-utils/fastapi-utils-0.2.1.tar.gz"
+
+    version("0.2.1", sha256="0e6c7fc1870b80e681494957abf65d4f4f42f4c7f70005918e9181b22f1bd759")
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi-utils/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on("py-poetry@0.12:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi-utils/package.py:19]: Missing deps:
```toml
fastapi = "*"
pydantic = "^1.0"
sqlalchemy = "^1.3.12"
```

File: var/spack/repos/builtin/packages/py-fastapi/package.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    pypi = "fastapi/fastapi-0.88.0.tar.gz"
+
+    version("0.88.0", sha256="915bf304180a0e7c5605ec81097b7d4cd8826ff87a02bb198e336fb9f3b5ff02")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-hatchling", type="build")
+
+    depends_on("py-starlette@0.22.0", type=("build", "run"))
+    depends_on("py-pydantic@1.6.2,1.7.4,1.8.2:1.9.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi/package.py]:
```suggestion
    depends_on("py-pydantic@1.6.2:1.6,1.7.4:1.7,1.8.2:1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-fastapi/package.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+class PyFastapi(PythonPackage):
+    """FastAPI framework, high performance, easy to learn, fast to code, ready for production"""
+
+    homepage = "https://github.com/tiangolo/fastapi"
+    pypi = "fastapi/fastapi-0.88.0.tar.gz"
+
+    version("0.88.0", sha256="915bf304180a0e7c5605ec81097b7d4cd8826ff87a02bb198e336fb9f3b5ff02")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-gunicorn/package.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+class PyGunicorn(PythonPackage):
+    """WSGI HTTP Server for UNIX"""
+
+    homepage = "https://gunicorn.org"
+    pypi = "gunicorn/gunicorn-20.1.0.tar.gz"
+
+    version("20.1.0", sha256="e0a968b5ba15f8a328fdfd7ab1fcb5af4470c28aaf7e55df02a99bc13138e6e8")
+
+    depends_on("python@3.5:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gunicorn/package.py]:
```suggestion
    depends_on("py-setuptools@3:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-gxformat2/package.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    """Galaxy Workflow Format 2 Descriptions"""
+
+    homepage = "https://github.com/galaxyproject/gxformat2"
+    pypi = "gxformat2/gxformat2-0.16.0.tar.gz"
+
+    version("0.16.0", sha256="16ff5aae1456e0a65c1ed644537e66626ea047a567c8eda19d2a2eb0b20fb752")
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gxformat2/package.py:21]: Can you add a comment saying that the tarball is missing the `requirements.txt` file but you can find it on GitHub?

File: var/spack/repos/builtin/packages/py-lagom/package.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+from spack.package import *
+
+
+class PyLagom(PythonPackage):
+    """Lagom is a dependency injection container
+    designed to give you 'just enough' help with building your dependencies.
+    """
+
+    homepage = "https://lagom-di.readthedocs.io"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lagom/package.py:14]: If there's no source code on PyPI can you install from GitHub? https://github.com/meadsteve/lagom/tags has the URLs. Then we don't need all this Python version-specific complexity.

File: var/spack/repos/builtin/packages/py-parsley/package.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+
+from spack.package import *
+
+
+class PyParsley(PythonPackage):
+    """Parsing and pattern matching made easy."""
+
+    homepage = "https://launchpad.net/parsley"
+    pypi = "parsley/Parsley-1.3.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-parsley/package.py]:
```suggestion
    pypi = "Parsley/Parsley-1.3.tar.gz"
```

File: var/spack/repos/builtin/packages/py-paste/package.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+
+from spack.package import *
+
+
+class PyPaste(PythonPackage):
+    """Tools for using a Web Server Gateway Interface stack"""
+
+    homepage = "https://pythonpaste.readthedocs.io"
+    pypi = "paste/Paste-3.5.2.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-paste/package.py]:
```suggestion
    pypi = "Paste/Paste-3.5.2.tar.gz"
```

File: var/spack/repos/builtin/packages/py-paste/package.py [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+
+    version("3.5.2", sha256="d5a7340c30bcdf3023dd0106c8a5c430dd8fe84aeb8113bc7b93f8dd729f4af6")
+
+    depends_on("python@3:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-paste/package.py]:
```suggestion
    depends_on("py-setuptools", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pastedeploy/package.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+    homepage = "https://docs.pylonsproject.org/projects/pastedeploy"
+    pypi = "PasteDeploy/PasteDeploy-3.0.1.tar.gz"
+
+    version("3.0.1", sha256="5f4b4d5fddd39b8947ea727161e366bf55b90efc60a4d1dd7976b9031d0b4e5f")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools@41:", type="build")
+
+    depends_on("py-importlib-metadata", when="^python@:3.7.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pastedeploy/package.py]:
```suggestion
    depends_on("py-importlib-metadata", when="^python@:3.7", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pebble/package.py [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+
+from spack.package import *
+
+
+class PyPebble(PythonPackage):
+    """Threading and multiprocessing eye-candy."""
+
+    homepage = "https://github.com/noxdafox/pebble"
+    pypi = "pebble/Pebble-5.0.3.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pebble/package.py]:
```suggestion
    pypi = "Pebble/Pebble-5.0.3.tar.gz"
```

File: var/spack/repos/builtin/packages/py-starlette/package.py [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    """The little ASGI library that shines."""
+
+    homepage = "https://github.com/encode/starlette"
+    pypi = "starlette/starlette-0.23.1.tar.gz"
+
+    version("0.23.1", sha256="8510e5b3d670326326c5c1d4cb657cc66832193fe5d5b7015a51c7b1e1b1bf42")
+    version("0.22.0", sha256="b092cbc365bea34dd6840b42861bdabb2f507f8671e642e8272d2442e08ea4ff")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-starlette/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-starlette/package.py [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-hatchling", type="build")
+
+    depends_on("py-anyio@3.4.0:4.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-starlette/package.py]:
```suggestion
    depends_on("py-anyio@3.4:4", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-starlette/package.py [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-typing-extensions@3.10.0:", when="^python@:3.9.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-starlette/package.py]:
```suggestion
    depends_on("py-typing-extensions@3.10.0:", when="^python@:3.9", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-gxformat2/package.py (hunk 9 above)
Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/py-gxformat2/package.py:21]: done!

File: var/spack/repos/builtin/packages/py-lagom/package.py (hunk 10 above)
Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/py-lagom/package.py:14]: done!

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py (hunk 5 above)
Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/py-fastapi-utils/package.py:19]: done!

**Comment by marcodelapierre** [comment]
Hi @adamjstewart, happy new year!

Here is an update:
1- requested changes implemented
2- latest develop merged in
3- CI checks ok

File: var/spack/repos/builtin/packages/py-dnspython/package.py [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+
+class PyDnspython(PythonPackage):
+    """DNS toolkit"""
+
+    homepage = "https://www.dnspython.org"
+    pypi = "dnspython/dnspython-2.2.1.tar.gz"
+
+    version("2.2.1", sha256="0f7569a4a6ff151958b64304071d370daa3243d15941a7beedf0c9fe5105603e")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dnspython/package.py]:
```suggestion
    depends_on("python@3.6:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+class PyFastapiUtils(PythonPackage):
+    """Reusable utilities for FastAPI"""
+
+    homepage = "https://fastapi-utils.davidmontague.xyz"
+    pypi = "fastapi-utils/fastapi-utils-0.2.1.tar.gz"
+
+    version("0.2.1", sha256="0e6c7fc1870b80e681494957abf65d4f4f42f4c7f70005918e9181b22f1bd759")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi-utils/package.py]:
```suggestion
    depends_on("python@3.6:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+    depends_on("py-poetry@0.12:", type="build")
+
+    depends_on("py-fastapi", type=("build", "run"))
+    depends_on("py-pydantic@1.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi-utils/package.py]:
```suggestion
    depends_on("py-pydantic@1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+    depends_on("py-sqlalchemy@1.3.12:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fastapi-utils/package.py]:
```suggestion
    depends_on("py-sqlalchemy@1.3.12:1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-lagom/package.py [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+    url = "https://github.com/meadsteve/lagom/archive/refs/tags/2.2.0.tar.gz"
+    git = "https://github.com/meadsteve/lagom.git"
+
+    version("2.2.0", sha256="69f701a2f81d9ca0ea7c93a5b15f7420bbe03d14175ec128959ad82e2b67460b")
+
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lagom/package.py]:
```suggestion
    depends_on("py-setuptools@59.6:", type="build")
    depends_on("py-mypy@0.971", type="build")
```

File: var/spack/repos/builtin/packages/py-supervisor/package.py [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+class PySupervisor(PythonPackage):
+    """A system for controlling process state under UNIX"""
+
+    homepage = "http://supervisord.org"
+    pypi = "supervisor/supervisor-4.2.4.tar.gz"
+
+    version("4.2.4", sha256="40dc582ce1eec631c3df79420b187a6da276bbd68a4ec0a8f1f123ea616b97a2")
+
+    depends_on("python@3.4:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-supervisor/package.py]:
```suggestion
    depends_on("py-setuptools", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-fastapi-utils/package.py (hunk 20 above)
Comment (marcodelapierre) [reply var/spack/repos/builtin/packages/py-fastapi-utils/package.py:18]: duplicate of one above

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by marcodelapierre** [comment]
@spackbot run pipeline

**Comment by marcodelapierre** [comment]
NOTE the merge is being blocked by a single failed test by `ci/gitlab-ci`: 

https://gitlab.spack.io/spack/spack/-/jobs/5109624

.. with a silly error, which I am unsure whether related or not with this specific PR:
```
WARNING: Failed to pull image with policy "IfNotPresent": image pull failed: pull QPS exceeded
ERROR: Job failed: prepare environment: waiting for pod running: pulling image "ghcr.io/spack/e4s-amazonlinux-2:v2022-03-21": image pull failed: pull QPS exceeded. Check https://docs.gitlab.com/runner/shells/index.html#shell-profile-loading for more information
```

Can this be fixed somehow?

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request merged**